docs/source/developers/release.rst

```diff
@@ -200,6 +200,8 @@ Build source and binaries and submit them
 .. code-block::
 
     # Build the source release tarball and create Pull Request with verification tasks
+    #
+    # Note that you need to have GitHub CLI installed to run this script.
```

Comment:
Should we add a note here saying that we have to wait for the workflow to finish creating the Release on GH before executing this? I usually execute those two scripts one after the other and if there is a minor queue on the runners and the release is not created this execution will fail to download the `tar.gz`. Adding a note to the release manager to validate the release is created could help prevent that.

Reply:
Will do!

Reply:
FYI: We can wait for the workflow in our script like ADBC does: https://github.com/apache/arrow-adbc/blob/ddb1bcf7dfb249dc3e958678bf4d65bd205ca58f/dev/release/02-sign.sh#L37-L51

Reply:
This is cool :) :+1:

Reply:
Thanks for the suggestion @kou! I added a new script called `dev/release/utils-watch-gh-workflow.sh`. I modified `02-source.sh` to call this new script so that it only attempts to download the tarball from the GitHub Release once the `release_candidate.yml` workflow is finished. Additionally, `post-05-update-gh-release-notes.sh` calls `dev/release/utils-watch-gh-workflow.sh` to wait for the `release.yml` workflow to finish before attempting to update the release notes.